trieste/models/keras/layers.py

```diff
@@ -0,0 +1,76 @@
+from typing import Union
+
+import tensorflow as tf
+from tensorflow.keras.layers import Dense, Dropout
+from tensorflow.python.ops import nn_ops, math_ops, sparse_ops, embedding_ops, gen_math_ops, standard_ops
+from tensorflow.python.framework import sparse_tensor
+from tensorflow.python.eager import context
+
+
+
+class DropConnect(Dense):
+    def __init__(self, rate: Union[float, int]=0.5, *args, **kwargs):
+        """
+        :param units: Number of units to use in the layer.
+        :param rate: The probability of dropout applied to each weight of a Dense Keras layer.
+        :param *args: Args passed to Dense Keras class.
+        "param **kwargs: Keyword arguments passed to Dense Keras class
+        """
+        self.rate = rate
+        super(DropConnect, self).__init__(*args, **kwargs)
+    
+    @property
+    def rate(self):
+        return self._rate
+
+    @rate.setter
+    def rate(self, rate):
+        assert 0 <= rate <= 1, f"prob needs to be a valid probability instead got {rate}"
```

Comment:
you should avoid using asserts in code, they are meant for tests, use tf.debugging for checking things